File: Processes/C++/wait_example.cpp
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <iostream>
#include <cstdlib>

int main() {
    pid_t pid = fork();
    if (pid < 0) {
        std::cerr << "fork failed" << std::endl;
        return 1;
    } else if (pid == 0) {
        sleep(1);
        exit(42);
    } else {
        int status;
        waitpid(pid, &status, 0);
        if (WIFEXITED(status)) {
            std::cout << "Child exited with status " << WEXITSTATUS(status) << std::endl;
        }
    }
    return 0;
}

File: Processes/Python/wait_example.py
import os
import sys

pid = os.fork()
if pid < 0:
    print("fork failed", file=sys.stderr)
    sys.exit(1)
elif pid == 0:
    os._exit(42)
else:
    pid_wait, status = os.wait()
    print(f"Child {pid_wait} exited with status {os.WEXITSTATUS(status)}")